scipy/sparse/linalg/_eigen/lobpcg/tests/test_lobpcg.py

```diff
@@ -371,19 +453,25 @@ def test_tolerance_float32():
     assert_allclose(eigvals, -np.arange(1, 1 + m), atol=2e-5, rtol=1e-5)
 
 
-def test_random_initial_float32():
-    """Check lobpcg in float32 for specific initial.
+@pytest.mark.parametrize("vdtype", VDTYPES)
+@pytest.mark.parametrize("mdtype", MDTYPES)
+@pytest.mark.parametrize("arr_type", [np.array,
+                                      sparse.csr_matrix,
+                                      sparse.coo_matrix])
+@pytest.mark.filterwarnings("ignore:Inplace")
+def test_dtypes(vdtype, mdtype, arr_type):
+    """Test lobpcg in various dtypes.
```

Comment:
This looks really nice!

Can we remove the filterwarnings (by removing `verbosityLevel=1` below)?

Reply:
> This looks really nice!
> 
> Can we remove the filterwarnings (by removing `verbosityLevel=1` below)?

The remarkable fact is that the "Inpace" warning is now back issued when real and complex data are mixed, like x is real and y is complex - unsure how this passes the check of the dtypes. So I can put back the test for the warning itself, although it would be nicer to find out how come it still gets issued.

Reply:
Yes, I will.